corporate/lib/stripe.py

```diff
@@ -317,6 +318,26 @@ def do_create_stripe_customer(user: UserProfile, stripe_token: Optional[str] = N
     return customer
 
 
+def do_create_zulip_sponsorship_request(
+    realm: Realm,
+    requested_by: UserProfile,
+    org_website: str,
+    org_description: str,
+    org_type: Optional[int] = Realm.ORG_TYPES["unspecified"]["id"],
+) -> ZulipSponsorshipRequest:
+    with transaction.atomic():
```

Comment:
This does not do anything.

I would actually inline this into its caller and `update_sponsorship_status` and have the `transaction.atomic` wrap the 3-4 database operations in sequence there.